airbyte-ci/connectors/connectors_qa/connectors_qa/checks/documentation.py

```diff
@@ -0,0 +1,206 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+import textwrap
+from typing import List
+
+from connector_ops.utils import Connector  # type: ignore
+from connectors_qa import consts
+from connectors_qa.models import Check, CheckCategory, CheckResult
+from pydash.objects import get  # type: ignore
+
+
+class DocumentationCheck(Check):
+    category = CheckCategory.DOCUMENTATION
+
+
+class CheckMigrationGuide(DocumentationCheck):
+    name = "Breaking changes must be accompanied by a migration guide"
+    description = "When a breaking change is introduced we check that a migration guide is available. It should be stored under  `./docs/integrations/<connector-type>s/<connector-name>-migrations.md`.\nThis document should contain a section for each breaking change, in order of the version descending. It must explain users which action to take to migrate to the new version."
+
+    def _run(self, connector: Connector) -> CheckResult:
+        breaking_changes = get(connector.metadata, "releases.breakingChanges")
+        if not breaking_changes:
+            return self.create_check_result(
+                connector=connector,
+                passed=True,
+                message="No breaking changes found. A migration guide is not required",
+            )
+        migration_guide_file_path = connector.migration_guide_file_path
+        migration_guide_exists = migration_guide_file_path is not None and migration_guide_file_path.exists()
+        if not migration_guide_exists:
+            return self.create_check_result(
+                connector=connector,
+                passed=False,
+                message=f"Migration guide file is missing for {connector.technical_name}. Please create a migration guide in ./docs/integrations/<connector-type>s/<connector-name>-migrations.md`",
+            )
+
+        expected_title = f"# {connector.name_from_metadata} Migration Guide"
+        expected_version_header_start = "## Upgrading to "
```

Comment:
Do we have all the critical rules covered?

Reply:
I ported over what currently exists in [`qa_checks.py` ](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/connector_ops/connector_ops/qa_checks.py). Let me know if you think additional checks should be implemented. I'm open for it, but in different PRs and with related new GH issues 😄